File: Cpp/src/array/richest_customer_wealth.cpp

#include "richest_customer_wealth.h"

using namespace std;

int RichestCustomerWealth::MaximumWealth(const vector<vector<int>> &accounts) {
    int result = 0;
    int current_customer_wealth = 0;
    for (const auto &account: accounts) // access by reference to avoid copying
    {
        current_customer_wealth = 0;
        for (const auto balance: account) {
            current_customer_wealth += balance;
        }
        if (current_customer_wealth > result) {
            result = current_customer_wealth;
        }
    }
    return result;
}

File: Python/src/array/richest_customer_wealth.py
from typing import List


class RichestCustomerWealth:

    def maximum_wealth(self, accounts: List[List[int]]) -> int:
        result = 0
        for resources in accounts:
            current_customer_wealth = 0
            for resource in resources:
                current_customer_wealth += resource
            result = max(current_customer_wealth, result)
        return result
